File: src/main/java/br/ufpb/dcx/project/si/service/UserService.java
package br.ufpb.dcx.project.si.service;

import br.ufpb.dcx.project.si.domain.User;
import br.ufpb.dcx.project.si.rest.config.EventException;

public interface UserService {

	User createUser(User user) throws EventException;
	
	User getUser(Integer userId) throws EventException;
	
	User updateUser(Integer userId, User user) throws EventException;
	
}

File: src/main/webapp/src/app/components/user/user.service.js
(function () {
  'use strict';

  angular
    .module('project-si')
    .service('UserService', UserService);

  function UserService($http, apiAddress) {

    var createUser = function (user) {
      return $http.post(apiAddress + "/user", user);
    };

    var getUser = function (userId) {
      return $http.get(apiAddress + "/user/" + userId);
    };

    var updateUser = function (userId, user) {
      return $http.put(apiAddress + "/user/" + userId, user);
    };

    return {
      createUser: createUser,
      getUser: getUser,
      updateUser: updateUser
    }


  }

})();
